modules/proxy-certificates.adoc

```diff
@@ -0,0 +1,114 @@
+
+// Module included in the following assemblies:
+//
+// * authentication/understanding-certificates.adoc
+
+[id="proxy-certificates_{context}"]
+= Proxy certificates
+
+== Purpose
+
+The `trustedCA` field of `proxy` is a reference to a ConfigMap that contains a
+user-provided trusted certificate authority (CA) bundle. This bundle is merged
+with the Red Hat Enterprise Linux CoreOS (RHCOS) trust bundle and injected into
+the trust store of platform components that make egress HTTPS calls. For
+example, `image-registry-operator` calls an external image registry to download
+images. If `trustedCA` is not specified, only the RHCOS trust bundle is used for
+proxied HTTPS connections. Provide custom CA certs to the RHCOS trust bundle if
+you want to use your own certificate infrastructure.
+
+The `trustedCA` field should only be consumed by a proxy validator. The
+validator is responsible for reading the certificate bundle from required key
+`ca-bundle.crt` and copying it to a ConfigMap named `trusted-ca-bundle` in the
+`openshift-config-managed` namespace. The namespace for the ConfigMap referenced
+by `trustedCA` is `openshift-config`:
+
+[source,yaml]
+----
+apiVersion: v1
+	kind: ConfigMap
+ metadata:
+	 name: user-ca-bundle
+	 namespace: openshift-config
+	 data:
+	   ca-bundle.crt: |
+	     -----BEGIN CERTIFICATE-----
+	     Custom CA certificate bundle.
+	     -----END CERTIFICATE-----
+----
+
+== File path
+
+The user-provided trust bundle is represented as a ConfigMap. The ConfigMap is
+mounted into the filesystem of platform components that make egress HTTPS calls.
+Typically, Operators mount the ConfigMap to
+`/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, but this not required by
+proxy and is at the Operator's discretion.
+
+If using the RHCOS trust bundle, place CA certificates in
+`/etc/pki/ca-trust/source/anchors`.
+
+== Expiration
+
+The user sets the expiration term of the user-provided trust bundle.
+
+The default expiration term is defined by the CA certificate itself. It is up to
+the CA administrator to configure this for the certificate before it can be used
+by {product-title} or RHCOS.
+
+== Services that use it
+
+When proxy is enabled, all platform components that make egress HTTPS calls will
+use either:
+
+* the merged user/system trust bundle if `trustedCA` is specified, or
+* the RHCOS trust bundle if `trustedCA` is not specified.
```

Comment:
How does a user know when which trust bundle will be used? 

In short, I don't think we should phrase this, stanza, this way. We need to be more explicit in telling a user when and which trusted CA will be used.

Reply:
@danehans Can you please help provide clarity here?

Reply:
@danehans Can you please comment?

Reply:
@sferich888 I've read through this section a couple times and I don't understand the confusion. In summary, the RHCOS trust bundle is used when `proxy.spec.trustedCA` is undefined and the merged user/system bundle is used when `proxy.spec.trustedCA` is defined.

Reply:
The wording leaves me wondering whether "system bundle" and "RHCOS bundle" re the same thing. The description of trustedCA suggests that it is ("This bundle is merged with the Red Hat Enterprise Linux CoreOS (RHCOS) trust bundle"). Ideally both conditions would use the same wording to refer to the same thing, such as:

```
* the merged user/RHCOS system trust bundle if `trustedCA` is specified, or
* the RHCOS system trust bundle if `trustedCA` is not specified.
```

Or to say:

```
When proxy is enabled all platform components that make egress HTTPS calls will
use the RHCOS trust bundle. If `trustedCA` is defined, it will also be used.
```

This avoids putting "use the RHCOS trust bundle" in both paths of the verbal conditional.

(Or maybe "system trust bundle" and "RHCOS trust bundle" really are two different things, in which case *that* should be clarified.)